include/cell.h

```diff
@@ -274,9 +277,19 @@ class Cell {
   std::map<unsigned, std::vector<std::pair<unsigned, double>>>
       velocity_constraints_;
 
-  //! Normal of face
-  //! first-> face_id, second->vector of the normal
-  std::map<unsigned, Eigen::VectorXd> face_normals_;
+  //! Parallel of face
+  //! first-> face_id, second->new coordinate axes
+  std::map<unsigned, Eigen::MatrixXd> new_coordinate_axes_;
+
+  //! Constrained nodal velocity
+  std::map<unsigned, Eigen::VectorXd> constrained_nodal_velocity_;
```

Comment:
You already have `velocity_constraints_`, why do you need these two? why can't be local to the apply velocity constraints function?